lib/ts/Arena.h

```diff
@@ -166,4 +167,34 @@ Arena::str_store(const char *str, size_t len)
   return mem;
 }
 
-#endif /* __ARENA_H__ */
+struct JemallocArena : public std::allocator<uint64_t>
+{
+  void *alloc(size_t size) { return allocate((size+1)/sizeof(uint64_t)); }
+  void free(void *mem, size_t size) { deallocate(static_cast<uint64_t*>(mem), size); }
+
+  char *str_alloc(size_t len) { return static_cast<char*>(alloc(len+1)); }
+  char *str_store(const char *str, size_t len)
+  {
+     char *mem = str_alloc(len);
+     memcpy(mem, str, len);
+     mem[len] = '\0';
+     return mem;
+  }
+
+  void str_free(char *str) { free(str,0); }
+
+  // does it the long hard way
+  static size_t str_length(const char *str) 
```

Comment:
I am unclear on why we have all these new functions. 

Adding a memory allocator is more about replacing new/malloc and delete/free backends. For example, this function does not seem to me that it should exist. strlen is a compiler intrinsic